exporter/kafkaexporter/exporter.go

```diff
@@ -0,0 +1,122 @@
+package kafkaexporter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/IBM/sarama"
+	"github.com/thomaspoignant/go-feature-flag/exporter"
+	"github.com/thomaspoignant/go-feature-flag/utils/fflog"
+)
+
+const (
+	formatJSON = "json"
+)
+
+// MessageSender is a Kafka producer that implements the SendMessages method
+type MessageSender interface {
+	SendMessages(msgs []*sarama.ProducerMessage) error
+}
+
+// Settings contains Kafka-specific configurations needed for message creation
+type Settings struct {
+	Topic     string
+	Addresses []string
+	*sarama.Config
+}
+
+// Exporter sends events to a Kafka topic using a synchronous producer
+type Exporter struct {
+	// Format is the output format for the message value.
+	// The only available format right now is JSON, and this field provided for future usage.
+	// Default: JSON
+	Format string
+
+	// Settings contains the Kafka producer's configuration. The Topic and Addresses fields are required. If
+	// no sarama.Config is provided a sensible default will be used.
+	Settings Settings
+
+	init   sync.Once
+	sender MessageSender
+	// dialer will create the producer. This field is added for dependency injection during testing as sarama
+	// has the annoying tendency to dial as soon as a producer is created.
+	dialer func(addrs []string, config *sarama.Config) (MessageSender, error)
+}
+
+// Export will produce a message to the Kafka topic. The message's value will contain the event encoded in the
+// selected format. Messages are publish synchronously and will error immediately on failure.
+func (e *Exporter) Export(_ context.Context, logger *log.Logger, featureEvents []exporter.FeatureEvent) error {
+	if e.sender == nil {
+		err := e.initializeProducer()
+		if err != nil {
+			return fmt.Errorf("writer: %w", err)
+		}
+	}
```

Comment:
I had an issue with the initialization here.
During my test what I've done is try to configure the exporter like this: 

```go
DataExporter: ffclient.DataExporter{
  FlushInterval:    1 * time.Second,
  MaxEventInMemory: 100,
  Exporter: &kafkaexporter.Exporter{
    Settings: kafkaexporter.Settings{},
  },
},
```

And what happened:
- The first call to the Kafka exporter went well because the `e.initializeProducer()` return an error
- But the second call `panic` because of the `e.init.Do` it is not throwing any error and we try to send a message with `e.sender.SendMessages(messages)` when `e.sender` is `nil`.

Reply:
I've replicated the issue now, my testing setup did not catch this previously, sorry. What would be the preferred behavior for calling the exporter after a failure? Should I continue to retry the connection?